tiledb/cf/engines/netcdf4_engine.py

```diff
@@ -427,94 +598,31 @@ def copy(
-                        except KeyError as err:
-                            raise KeyError(
-                                f"Variable {var_converter.input_name} not found in "
-                                f"requested NetCDF group."
-                            ) from err
-                        data[var_converter.output_name] = variable[...]
-                        attr_meta = group.get_attr_metadata(var_converter.output_name)
-                        for meta_key in variable.ncattrs():
-                            copy_metadata_item(attr_meta, variable, meta_key)
-                    dim_slice = tuple(slice(dim.size) for dim in variable.get_dims())
-                    group.array[dim_slice or slice(None)] = data
-
-    def rename_array(self, original_name: str, new_name: str):
-        """Renames an array in the output TileDB CF dataspace.
-
-        Parameters:
-            original_name: Current name of the array to be renamed.
-            new_name: New name the array will be renamed to.
-        """
-        if new_name in self._arrays:
-            raise ValueError(
-                f"Cannot rename array '{original_name}' to '{new_name}'. An array with "
-                f"that name already exists."
-            )
-        super().rename_array(original_name, new_name)
-        self._arrays[new_name] = self._arrays.pop(original_name)
-
-    def rename_attr(self, original_name: str, new_name: str):
-        """Renames an attribute in the output TileDB CF dataspace.
-
-        Parameters:
-            original_name: Current name of the attribute to be renamed.
-            new_name: New name the attribute will be renamed to.
-        """
-        if new_name in self._variables:
-            raise ValueError(
-                f"Cannot rename attribute '{original_name}' to '{new_name}'. An "
-                f"attribute with that name already exists."
-            )
-        super().rename_attr(original_name, new_name)
-        self._variables[new_name] = self._variables.pop(original_name)
-        self._variables[new_name].output_name = new_name
-
-    def rename_dim(self, original_name: str, new_name: str):
-        """Renames a dimension in the output TileDB CF dataspace.
-
-        Parameters:
-            original_name: Current name of the dimension to be renamed.
-            new_name: New name the dimension will be renamed to.
-        """
-        if new_name in self._dimensions:
-            raise ValueError(
-                f"Cannot rename dimension '{original_name}' to '{new_name}'. A "
-                f"dimension with that name already exists."
-            )
-        super().rename_dim(original_name, new_name)
-        self._dimensions[new_name] = self._dimensions.pop(original_name)
-        self._dimensions[new_name].output_name = new_name
+            for array_name, array_creator in self._array_creators.items():
+                if isinstance(array_creator, NetCDFArrayConverter):
```

Comment:
Can `array_creator` be anything else?

Reply:
Not currently, but I plan on adding support for non-converter arrays.